### PR TITLE
Feature/nijo 691 counterparty management

### PR DESCRIPTION
**Pull request by hasmikharutyunyan**

### CODE REVIEW

File: Api/Services/AdministratorServices/CounterpartyManagementService.cs [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+                    counterpartyToUpdate.PostalCode,
+                    counterpartyToUpdate.PreferredCurrency,
+                    counterpartyToUpdate.PreferredPaymentMethod,
+                    counterpartyToUpdate.Website,
+                    counterpartyToUpdate.VatNumber));
+            }
+        }
+
+
+        //This method is the same with CounterPartyService.GetCounterParty,
```
Comment (mingazhev) [inline Api/Services/AdministratorServices/CounterpartyManagementService.cs:98]: Please add whitespaces after "//"

File: Api/Services/Agents/ICounterpartyService.cs [hunk 2]
```diff
@@ -13,18 +13,12 @@ public interface ICounterpartyService
...
         Task<Result<AgencyInfo>> GetAgency(int agencyId);
 
-        Task<Result<List<AgencyInfo>>> GetAllCounterpartyAgencies(int counterpartyId);
-
         Task<Agency> GetDefaultAgency(int counterpartyId);
 
-        Task<Result> VerifyAsFullyAccessed(int counterpartyId, string verificationReason);
-
-        Task<Result> VerifyAsReadOnly(int counterpartyId, string verificationReason);
+
```
Comment (mingazhev) [inline Api/Services/Agents/ICounterpartyService.cs:22]: Redundant blank line

File: Api/Services/AdministratorServices/CounterpartyManagementService.cs [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+            => Result.Ok(await _context.Counterparties.Select(counterparty => ToCounterPartyInfo(counterparty)).ToListAsync());
+
+
+        public Task<Result<List<AgencyInfo>>> GetAllCounterpartyAgencies(int counterpartyId)
+        {
+            return Get(counterpartyId)
+                .Bind(counterparty => GetAgencies());
+
+
+            async Task<Result<List<AgencyInfo>>> GetAgencies()
```
Comment (mingazhev) [inline Api/Services/AdministratorServices/CounterpartyManagementService.cs:47]: Could this be done withot Result<T>?

Comment (hasmikharutyunyan) [reply Api/Services/AdministratorServices/CounterpartyManagementService.cs:47]: Yes, but I think for compatibility will be better to return Result, but I can change

Comment (mingazhev) [reply Api/Services/AdministratorServices/CounterpartyManagementService.cs:47]: No need to return Result if result is always same. And async\await can be removed there too.

File: Api/Services/Agents/CounterPartyValidator.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+using CSharpFunctionalExtensions;
+using FluentValidation;
+using HappyTravel.Edo.Api.Infrastructure;
+using HappyTravel.Edo.Api.Models.Agents;
+
+namespace HappyTravel.Edo.Api.Services.Agents
+{
+    public static class CounterPartyValidator
```
Comment (mingazhev) [inline Api/Services/Agents/CounterPartyValidator.cs:8]: "Counterparty" is the whole word in our terms, not "Counter party" (check the letters' case)

File: Api/Services/AdministratorServices/CounterpartyManagementService.cs [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+            _managementAuditService = managementAuditService;
+            _permissionManagementService = permissionManagementService;
+            _accountManagementService = accountManagementService;
+        }
+
+
+        public async Task<Result<CounterpartyInfo>> Get(int counterpartyId) => await GetCounterparty(counterpartyId).Map(ToCounterPartyInfo);
+
+
+        public async Task<Result<List<CounterpartyInfo>>> Get()
```
Comment (mingazhev) [inline Api/Services/AdministratorServices/CounterpartyManagementService.cs:37]: Same there, this method never returns failed result

File: Api/Services/AdministratorServices/CounterpartyManagementService.cs (hunk 3 above)
Comment (hasmikharutyunyan) [reply Api/Services/AdministratorServices/CounterpartyManagementService.cs:47]: `GetAgencies` does not use parameter , I can change declaration of `GetAgencies`,but not use the parameter

File: Api/Controllers/AdministratorControllers/CounterpartiesController.cs [hunk 6]
```diff
@@ -19,12 +21,42 @@ namespace HappyTravel.Edo.Api.Controllers.AdministratorControllers
...
+        [ProducesResponseType(typeof(List<CounterpartyInfo>), (int) HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.BadRequest)]
+        [AdministratorPermissions(AdministratorPermissions.CounterpartyManagement)]
+        public async Task<IActionResult> Get(int counterpartyId)
+        {
+            var (_, isFailure, counterParties, error) = await _counterpartyManagementService.Get(counterpartyId);
+            if (isFailure)
+                return BadRequest(ProblemDetailsBuilder.Build(error));
+
+            return Ok(counterParties);
```
[the remark this answers is not in the document]
Comment (Raibo) [reply Api/Controllers/AdministratorControllers/CounterpartiesController.cs:45]: ps ставлю аппрув чтобы не надо было потом меня ждать, замечние очень простое, а других у меня нет

[the remark this answers is not in the document]
Comment (mingazhev) [reply Api/Controllers/AdministratorControllers/CounterpartiesController.cs:45]: @Raibo pls use English

**Pull request merged**